drift/lib/src/runtime/manager/manager.dart

```diff
@@ -292,53 +275,71 @@ class TableManagerState<
 /// This is so that the state can be passed down to lower level managers
+@immutable
 abstract class BaseTableManager<
-        DB extends GeneratedDatabase,
-        T extends Table,
-        DT,
-        FS extends FilterComposer<DB, T>,
-        OS extends OrderingComposer<DB, T>,
-        C extends ProcessedTableManager<DB, T, DT, FS, OS, C, CI, CU>,
-        CI extends Function,
-        CU extends Function>
-    implements
-        MultiSelectable<DT>,
-        SingleSelectable<DT>,
-        SingleOrNullSelectable<DT> {
+    $Database extends GeneratedDatabase,
+    $Table extends Table,
+    $Dataclass,
+    $FilterComposer extends FilterComposer<$Database, $Table>,
+    $OrderingComposer extends OrderingComposer<$Database, $Table>,
+    $CreateCompanionCallback extends Function,
+    $UpdateCompanionCallback extends Function> extends Selectable<$Dataclass> {
   /// The state for this manager
-  final TableManagerState<DB, T, DT, FS, OS, C, CI, CU> $state;
+  final TableManagerState<
+      $Database,
+      $Table,
+      $Dataclass,
+      $FilterComposer,
+      $OrderingComposer,
+      $CreateCompanionCallback,
+      $UpdateCompanionCallback> $state;
 
   /// Create a new [BaseTableManager] instance
   ///
   /// {@macro manager_internal_use_only}
-  const BaseTableManager(this.$state);
+  BaseTableManager(this.$state);
 
   /// Add a limit to the statement
-  C limit(int limit, {int? offset}) {
-    return $state
-        ._getChildManagerBuilder($state.copyWith(limit: limit, offset: offset));
+  ProcessedTableManager<
+      $Database,
+      $Table,
+      $Dataclass,
+      $FilterComposer,
+      $OrderingComposer,
+      $CreateCompanionCallback,
+      $UpdateCompanionCallback> limit(int limit, {int? offset}) {
+    return ProcessedTableManager($state.copyWith(limit: limit, offset: offset));
   }
 
   /// Add ordering to the statement
-  C orderBy(ComposableOrdering Function(OS o) o) {
+  ProcessedTableManager<$Database, $Table, $Dataclass, $FilterComposer,
+          $OrderingComposer, $CreateCompanionCallback, $UpdateCompanionCallback>
+      orderBy(ComposableOrdering Function($OrderingComposer o) o) {
     final orderings = o($state.orderingComposer);
-    return $state._getChildManagerBuilder($state.copyWith(
+    return ProcessedTableManager($state.copyWith(
         orderingBuilders:
             $state.orderingBuilders.union(orderings.orderingBuilders),
         joinBuilders: $state.joinBuilders.union(orderings.joinBuilders)));
   }
 
   /// Add a filter to the statement
-  C filter(ComposableFilter Function(FS f) f) {
+  ///
+  /// The [combineWith] parameter can be used to combine the new filter with any existing filters
+  ProcessedTableManager<$Database, $Table, $Dataclass, $FilterComposer,
+          $OrderingComposer, $CreateCompanionCallback, $UpdateCompanionCallback>
+      filter(ComposableFilter Function($FilterComposer f) f,
```

Comment:
I wonder if having two methods instead of the `combineWith` parameter may be easier?

```dart
        db.managers.tableWithEveryColumnType
            .filter((f) => f.aText("person"))
            .or((f) => f.aReal(5.0))
```

But then we kind of start mirroring the filter API in the general manager API which is also not too nice... Maybe we can also keep it as it is, what you do think?

Reply:
This is done in preparation for the references pull request.

There will need to combine managers, each with their own query, into 1 manager which has all the filters combined with an OR statement.

However, being that this is only needed internally, I'll make it private.